### PR TITLE
PMM-10445 Replace local reviewdog with github action version

### PR DESCRIPTION
**Pull request by Azanul**
This PR makes use of github-actions-version of reviewdog.

Resolves issue #1239

### CODE REVIEW

File: .github/workflows/common.yml [hunk 1]
```diff
@@ -80,22 +80,16 @@ jobs:
...
+      - name: golangci-lint
+        uses: reviewdog/action-golangci-lint@v2
+        with: 
+          github_token: ${{ secrets.ROBOT_TOKEN || secrets.GITHUB_TOKEN }}
+          go_version: "1.19"
+          reporter: github-pr-review
+          reviewdog_flags: "-f=golangci-lint"
+          fail_on_error: true
+          golangci_lint_flags: "-c=.golangci-required.yml --out-format=line-number"
+          golangci_lint_version: v1.46.2
```
Comment (BupycHuk) [inline .github/workflows/common.yml:92]: the current version that we use is 1.49.0
https://github.com/percona/pmm/blob/393d71c5be276fd683d0405938b8b4d35984ff4c/Makefile.include#L22

**Review by BupycHuk — COMMENTED** [review]
Hello @Azanul,
Thank you for you contribution, I left a few comments to improve the config.

File: .github/workflows/common.yml (hunk 1 above)
Comment (Azanul) [reply .github/workflows/common.yml]: Updated

**Review by BupycHuk — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

**Comment by artemgavrilov** [comment]
@Azanul Can you please do the same also for optional liners invocation: https://github.com/percona/pmm/blob/7e488993dee8ea67696e2c463a19f47f9dc9f4f6/.github/workflows/common.yml#L117-L132

Thank you!

**Comment by Azanul** [comment]
> @Azanul Can you please do the same also for optional liners invocation:
> 
> https://github.com/percona/pmm/blob/7e488993dee8ea67696e2c463a19f47f9dc9f4f6/.github/workflows/common.yml#L117-L132
> 
> Thank you!

**Comment by Azanul** [comment]
Tried to comment, closed by mistake

File: .github/workflows/common.yml [hunk 2]
```diff
@@ -115,21 +109,14 @@ jobs:
...
-          else
-            echo "$out"
-          fi
+        uses: reviewdog/action-golangci-lint@v2
+        with: 
+          github_token: ${{ secrets.ROBOT_TOKEN || secrets.GITHUB_TOKEN }}
+          go_version: ${{ env.GO_VERSION }}
+          reporter: github-pr-review
+          reviewdog_flags: "-f=golangci-lint"
+          golangci_lint_flags: "-c=.golangci-required.yml --out-format=line-number --issues-exit-code=0"
```
Comment (artemgavrilov) [inline .github/workflows/common.yml:118]:
```suggestion
          golangci_lint_flags: "-c=.golangci.yml --out-format=line-number --issues-exit-code=0"
```

File: .github/workflows/common.yml [hunk 3]
```diff
@@ -81,21 +81,15 @@ jobs:
...
-            fi
-          else
-            echo "$out"
-          fi
+        uses: reviewdog/action-golangci-lint@v2
+        with: 
+          github_token: ${{ secrets.ROBOT_TOKEN || secrets.GITHUB_TOKEN }}
+          go_version: ${{ env.GO_VERSION }}
+          reporter: github-pr-review
+          reviewdog_flags: "-f=golangci-lint"
```
Comment (artemgavrilov) [inline .github/workflows/common.yml:89]: I guess we can drop this flag. This action works with golangci-lint by default.

File: .github/workflows/common.yml [hunk 4]
```diff
@@ -115,21 +109,14 @@ jobs:
...
```
Comment (artemgavrilov) [inline .github/workflows/common.yml:117]: Same

**Review by artemgavrilov — APPROVED** [review]

**Pull request merged**